packages/amplify_datastore/android/build.gradle

```diff
@@ -0,0 +1,66 @@
+group 'com.amazonaws.amplify.amplify_datastore'
+version '1.0-SNAPSHOT'
+
+buildscript {
+    ext.kotlin_version = '1.3.50'
+    repositories {
+        google()
+        jcenter()
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    }
+}
+
+rootProject.allprojects {
+    repositories {
+        google()
+        jcenter()
+    }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdkVersion 28
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
+        test.java.srcDirs += 'src/test/kotlin'
+        test.resources.srcDirs += '../test/resources'
+    }
+
+    defaultConfig {
+        minSdkVersion 21
+    }
+    lintOptions {
+        disable 'InvalidPackage'
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
+}
+
+dependencies {
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.amplifyframework:aws-datastore:1.+"
+    implementation "com.amplifyframework:aws-api:1.+"
+    implementation "com.amplifyframework:aws-api-appsync:1.+"
+    testImplementation 'junit:junit:4.13'
+    testImplementation 'org.mockito:mockito-core:3.1.0'
+    testImplementation 'org.mockito:mockito-inline:3.1.0'
+    testImplementation 'androidx.test:core:1.2.0'
+    testImplementation 'org.robolectric:robolectric:4.3.1'
+    testImplementation 'com.fasterxml.jackson.core:jackson-core:2.6.3'
+    testImplementation 'com.fasterxml.jackson.core:jackson-annotations:2.6.3'
+    testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.6.3'
```

Comment:
Dang. Wish these said Gson.

Reply:
Me too, it was difficult to deserialize a number to an integer using Gson without providing extra deserialize adapters that would use the modelSchema. For the query tests (That don't use modelSchema at all) I felt it was overly complicated to achieve that with Gson, that's why relied on jackson for unit tests.